Sources/Datadog/RUM/AutoInstrumentation/Views/SwiftUIViewModifier.swift

```diff
@@ -0,0 +1,52 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+#if canImport(SwiftUI)
+import SwiftUI
+
+/// `SwiftUI.ViewModifier` for RUM which invoke `startView` and `stopView` from the
+/// global RUM Monitor when the modified view appears and disappears.
+@available(iOS 13, *)
+internal struct RUMViewModifier: SwiftUI.ViewModifier {
+    /// The Content View identifier.
+    let key: String
+
+    /// View name used for RUM Explorer. If not provided, the `key` name will be used.
+    let name: String?
+
+    /// Custom attributes to attach to the View.
+    let attributes: [AttributeKey: AttributeValue]
+
+    func body(content: Content) -> some View {
+        content.onAppear {
+            Global.rum.startView(key: key, name: name, attributes: attributes)
+        }
+        .onDisappear {
+            Global.rum.stopView(key: key, attributes: attributes)
+        }
+    }
+}
+
+@available(iOS 13, *)
+public extension SwiftUI.View {
+    /// Monitor this view with Datadog RUM. A start and stop events will be logged when this view appears
+    /// and disappears.
+    ///
+    /// - Parameters:
+    ///   - key: a `String` value identifying this View.
+    ///   - name: the View name used for RUM Explorer. If not provided, the `key` name will be used.
+    ///   - attributes: custom attributes to attach to the View.
+    /// - Returns: This view after applying a `ViewModifier` for monitoring the view.
+    func monitorView(
+        key: String,
+        name: String? = nil,
+        attributes: [AttributeKey: AttributeValue] = [:]
+    ) -> some View {
```

Comment:
I'm not sure on this signature. TO me, seeing `.monitorView(key: "SwiftUI View")` is quite puzzling at the first glance. It doesn't mention Datadog or RUM so it's hard to figure out where does this modifier come from. Also, with "monitor" being both a verb and noun this is a bit misleading. I know it was discussed in the RFC, but I intentionally skipped commenting there as it's an implementation detail to me.

Maybe `monitorRUMView(...)` or `trackRUMView(...)` (a counterpart to [`trackUIKitRUMViews(...)`](https://github.com/DataDog/dd-sdk-ios/blob/0d1fbd1cf2819576ba4a797d5184ede94d8e6555/Sources/Datadog/DatadogConfiguration.swift#L564)) could work better? We uniformly add `*RUM*` phrase to our APIs, IMO this could be preserved here as well.

Further, `key` parameter is just a technical reference for matching "start" and "stop" commands added for React-Native (where we can't identify RUM Views by `UIViewController` object reference). I don't think this makes sense to expose it in this manual API, as that would be forcing users to always give it unique value to avoid clashing views with common `name`. IMO this:
```swift
name: String,
attributes: [AttributeKey: AttributeValue] = [:]
```
can be enough. The name would be mandatory (makes sense for manual API) and we can manage `key` complexity inside `RUMViewModifier` implementation, e.g. by using `UUID()`:
```diff
- let key: String
+ private let key: String = UUID().uuidString
```

`String` is already a [supported `RUMViewIdentifiable` type](https://github.com/DataDog/dd-sdk-ios/blob/0d1fbd1cf2819576ba4a797d5184ede94d8e6555/Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewIdentity.swift#L49). To my knowledge, there is no Swift UI specific we could leverage for additional conformance, but if you have any idea, that could be an option as well.

Reply:
I see 👍. I'm sceptical to changing the `RUMMonitor` APIs signature, especially adding the `path` value back (it was removed in https://github.com/DataDog/dd-sdk-ios/pull/418). The `path` attribute is here for historical reason and was made for compatibility with the part of RUM schema coming form Browser SDK (it renders as `view.url` in web-ui - rather puzzling for mobile devs 📱). There was also an update to `path` inference from `key` in https://github.com/DataDog/dd-sdk-ios/pull/534.

Trying to distill the problem makes me came up with this constraints:
* (1) to make the API convenient, we want to ask as little values as possible (idealy: only `name`);
* (2) single value cannot solve clashing views problem (starting distinct `"Foo"` view after previous `"Foo"`);
* (3) referring to https://github.com/DataDog/dd-sdk-ios/pull/534 , the `path` must be a technical reference (independent from translations and `name` changes).

(1) and (2) can be fulfilled by making the `name` a non-optional parameter and adding optional `key` with default `nil` value. Under the hood we could use `name + (key ?? "")` for [`RUMViewIdentity`](https://github.com/DataDog/dd-sdk-ios/blob/0d1fbd1cf2819576ba4a797d5184ede94d8e6555/Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewIdentity.swift#L13). That might require using [`RUMCommandSubscriber`](https://github.com/DataDog/dd-sdk-ios/blob/0d1fbd1cf2819576ba4a797d5184ede94d8e6555/Sources/Datadog/RUM/AutoInstrumentation/RUMCommandSubscriber.swift#L9) instead of `RUMMonitor` to manage the RUM view.

The difficulty of (3) is defined by SwiftUI paradigm, where views have no identity (like `UIViewController` class in UIKit). If only we get something meaningful from `SwiftUI.View` (e.g. a `struct` name), that could work well for `path`. Is this sth that you mean?

>  I also thought about using the `SwiftUI.View's` composed body type

I think this might work 👌 - if we can indeed inspect the `content: Content` to pick some of its traits and call it a `path` (`view.url`) that would be similar to `path` being the `UIViewController's` class name in our `UIKit` monitoring.